### ISSUE TITLE
duckdb_append_blob returns DuckdDBError when blob data includes 0x00 character

### ISSUE THREAD

**Issue by suketa** [duckdb/duckdb#3960]
#### What happens?
`duckdb_append_blob` returns `DuckDBError` when data includes NUL (0x00) character in duckdb v0.4.0.

#### To Reproduce

```
char data[3];
data[0] = 0;
data[1] = 1;
data[2] = 2;

duckdb_append_blob(appender, (void *)data, 3); // => DuckDBError
```

In duckdb v0.3.4 duckdb_append_blob does not return DuckDBError.

#### Environment (please complete the following information):
 - OS: Ubuntu 20.04
 - DuckDB Version: 040
 
#### Identity Disclosure:
 - Full Name: Masaki Suketa
 - Affiliation: Private person. Author of  [ruby-duckdb](https://github.com/suketa/ruby-duckdb).

#### Before Submitting

- [x] **Have you tried this on the latest `master` branch?**
* **Other Platforms**: compile from source on Ubuntu 20.04

**Comment by Mytherin** [comment]
Thanks for the report! I can confirm there is a regression here - I have pushed a fix in #3961.

**Comment by calvin2021y** [comment]
I use v0.4.0. is there a way to work around this without build myself ?
